StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1203SA1214SA1215CodeFixProvider.cs

```diff
@@ -0,0 +1,101 @@
+
+        /// <inheritdoc/>
+        public override ImmutableArray<string> FixableDiagnosticIds => FixableDiagnostics;
+
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
+        {
+            return CustomFixAllProviders.BatchFixer;
+        }
+
+        /// <inheritdoc/>
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            foreach (Diagnostic diagnostic in context.Diagnostics.Where(d => FixableDiagnostics.Contains(d.Id)))
+            {
+                context.RegisterCodeFix(CodeAction.Create(OrderingResources.SA1203CodeFix, token => GetTransformedDocumentAsync(context.Document, diagnostic, token), equivalenceKey: nameof(SA1203SA1214SA1215CodeFixProvider)), diagnostic);
+            }
+
+            return SpecializedTasks.CompletedTask;
+        }
+
+        private static async Task<Document> GetTransformedDocumentAsync(Document document, Diagnostic diagnostic, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var fieldDeclaration = syntaxRoot.FindNode(diagnostic.Location.SourceSpan).FirstAncestorOrSelf<FieldDeclarationSyntax>();
+            var typeDeclarationNode = fieldDeclaration.FirstAncestorOrSelf<TypeDeclarationSyntax>();
+            if (typeDeclarationNode == null)
+            {
+                return document;
+            }
+
+            syntaxRoot = MoveField(fieldDeclaration, typeDeclarationNode, syntaxRoot);
+
+            return document.WithSyntaxRoot(syntaxRoot);
+        }
+
+        private static SyntaxNode MoveField(FieldDeclarationSyntax fieldDeclaration, TypeDeclarationSyntax typeDeclarationNode, SyntaxNode syntaxRoot)
+        {
+            var fieldToMove = new MemberOrderHelper(fieldDeclaration);
+            foreach (var member in typeDeclarationNode.Members)
+            {
+                if (!member.IsKind(SyntaxKind.FieldDeclaration))
+                {
+                    continue;
+                }
+
+                var orderHelper = new MemberOrderHelper(member);
+                if (orderHelper.ModifierPriority < fieldToMove.ModifierPriority)
+                {
+                    syntaxRoot = MoveField(syntaxRoot, fieldToMove.Member, member);
+                    break;
+                }
+            }
+
+            return syntaxRoot;
+        }
+
+        private static SyntaxNode MoveField(SyntaxNode root, MemberDeclarationSyntax field, MemberDeclarationSyntax firstNonConst)
+        {
+            var trackedRoot = root.TrackNodes(field, firstNonConst);
+            var fieldToMove = trackedRoot.GetCurrentNode(field);
+            var firstNonConstTracked = trackedRoot.GetCurrentNode(firstNonConst);
+            if (!fieldToMove.HasLeadingTrivia)
+            {
+                fieldToMove = fieldToMove.WithLeadingTrivia(firstNonConstTracked.GetLeadingTrivia());
```

Comment:
I tried this and it transforms this
```csharp
class Foo
{
    /// <summary>
    /// Bar
    /// </summary>
    string bar;
const string foo = "a";
}
```
to this
```csharp
class Foo
{
    /// <summary>
    /// Bar
    /// </summary>
    const string foo = "a";
    /// <summary>
    /// Bar
    /// </summary>
    string bar;
}
```

I would just not worry about trivia here.